scripts/generate_ipfs_hashes.py

```diff
@@ -0,0 +1,196 @@
+                    break
+
+        if len(fingerprints_dict) > 0:
+            replacement = "fingerprint:\n"
+            for file_name, ipfs_hash in fingerprints_dict.items():
+                replacement += "  " + file_name + ": " + ipfs_hash + "\n"
+        else:
+            replacement = "fingerprint: {}\n"
+        whole_file = whole_file.replace(existing, replacement)
+
+        # close the file
+        file.close()
+
+        # update fingerprints
+        with open(yaml_path, "w") as f:
+            f.write(whole_file)
+
+    # hash again to get outer hash (this time all files):
+    result_list = client.add(target_dir)
+    for result_dict in result_list:
+        if package_name == result_dict["Name"]:
+            key = os.path.join(AUTHOR, package_type, package_name)
+            package_hashes[key] = result_dict["Hash"]
+
+
+def to_csv(package_hashes: Dict[str, str]):
+    """Outputs a dictionary to CSV."""
+    try:
+        ordered = collections.OrderedDict(sorted(package_hashes.items()))
+        with open(PACKAGE_HASHES_PATH, "w") as csv_file:
+            writer = csv.writer(csv_file)
+            writer.writerows(ordered.items())
+    except IOError:
+        print("I/O error")
+
+
+if __name__ == "__main__":
+
+    # check we have ipfs
+    res = shutil.which("ipfs")
+    if res is None:
+        print("Please install IPFS first!")
+        sys.exit(1)
+
+    # run the ipfs daemon
+    process = subprocess.Popen(  # nosec
+        ["ipfs", "daemon"], stdout=subprocess.PIPE, env=os.environ.copy(),
+    )
+    time.sleep(2.0)
+
+    package_hashes = {}  # type: Dict[str, str]
+
+    try:
+        # connect ipfs client
+        client = ipfshttpclient.connect("/ip4/127.0.0.1/tcp/5001/http")
+        ipfs_hash_only = IPFSHashOnly()
+
+        # ipfs hash the core packages
+        for package_type, package_name in CORE_PACKAGES.items():
+            target_dir = os.path.join(CORE_PATH, package_type, package_name)
+            ipfs_hashing(
+                package_hashes, target_dir, package_type, package_name, ipfs_hash_only
+            )
+
+        # ipfs hash the registry packages
+        for package_type in PACKAGE_TYPES:
+            path = os.path.join(PACKAGE_PATH, package_type)
+            for (dirpath, dirnames, _filenames) in os.walk(path):
+                if dirpath.count("/") > 2:
+                    # don't hash subdirs
+                    break
+                for dirname in dirnames:
+                    target_dir = os.path.join(dirpath, dirname)
+                    ipfs_hashing(
+                        package_hashes,
+                        target_dir,
+                        package_type,
+                        dirname,
+                        ipfs_hash_only,
+                    )
+
+        # output the package hashes
+        to_csv(package_hashes)
+
+    except Exception as e:
+        print(e)
+
+    finally:
+        # terminate the ipfs daemon
```

Comment:
I would try to call `process.terminate()` (i.e. `SIGINT` on Posix OSs), and wait for some time before `process.kill()` (i.e. `SIGTERM` on Posix OSs).

https://docs.python.org/3/library/subprocess.html#subprocess.Popen.terminate

Reply:
Resolved it